sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -129,7 +129,7 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     val invalidKeys = table.properties.keys.filter(_.startsWith(SPARK_SQL_PREFIX))
     if (invalidKeys.nonEmpty) {
       throw new AnalysisException(s"Cannot persistent ${table.qualifiedName} into hive metastore " +
-        s"as table property keys may not start with '$SPARK_SQL_PREFIX': " +
+        s"as table property keys may start with '$SPARK_SQL_PREFIX': " +
```

Comment:
`User-specified table property keys should not start with '$SPARK_SQL_PREFIX'. Invalid table properties: `.

Reply:
But when I execute a DDL in spark-sql, still throw the analysis exception.
`spark-sql> ALTER TABLE gja_test3 SET TBLPROPERTIES ('test' = 'test');
org.apache.spark.sql.AnalysisException: Cannot persistent work.gja_test3 into hive metastore as table property keys may not start with 'spark.sql.': [spark.sql.partitionProvider];
at org.apache.spark.sql.hive.HiveExternalCatalog.verifyTableProperties(HiveExternalCatalog.scala:129)`
As you see, this ddl contains a user-specified table property key not start with '$SPARK_SQL_PREFIX'.

Reply:
So I think the message should change to `Table property keys should not start with '$SPARK_SQL_PREFIX'. Invalid table properties:`

Reply:
> `User-specified table property keys should not start with '$SPARK_SQL_PREFIX'. Invalid table properties: `.
There exists a detail is the analysis exception contains a word 'as', the meaning of 'as' is because. so I still think `as table property keys may start with '$SPARK_SQL_PREFIX' is better.

Reply:
> `User-specified table property keys should not start with '$SPARK_SQL_PREFIX'. Invalid table properties: `.
Please review this pr again,thanks!

Reply:
How did you create such a table?

Reply:
Could you show how you hit this error? It sounds like a bug to me.

Reply:
First, I use spark to create a managed table, and then to alter the table with set properties.
I guess this is a bug too!

Reply:
If you think this is a bug, I will fix it.

Reply:
I maintain a spark have a lot of custom function. My spark doesn't treat the table properties correctly lead to the analysis exception.
I have test the spark2.3 and spark2.4,this analysis exception not appear.
So we should back to the beginning, to discuss the incorrect analysis exception.
There exists a detail is the analysis exception contains a word 'as', the meaning of 'as' is because. so I still think `as table property keys may start with '$SPARK_SQL_PREFIX'` is better.

Reply:
> Could you show how you hit this error? It sounds like a bug to me.
Please review this pr again,thanks!

Reply:
I don't understand this change; it's checking whether table properties start with this prefix, and it's an error if they do. The error said it's not allowed before, but now it says it is. Isn't that reversed?

Do you mean something like, internal keys that start with this prefix are reserved, and user-specified keys may not start with the prefix? but I think the existing message already communicated the latter part.

Reply:
Yes, only internal keys allowed start with this prefix and user-specified keys not allowed with this prefix.If user-specified keys start with this prefix, throw the AnalysisException so that tell the reason.
The reason is user-specified keys start with this prefix.
The reason isn't user-specified keys not start with this prefix.

Reply:
@srowen There exists a detail is internal keys start with this prefix has been filtered out before call method `verifyTableProperties`.

Reply:
User-specified keys start with this prefix not filtered out before call method `verifyTableProperties`.